apps/rush-lib/src/logic/Git.ts

```diff
@@ -136,33 +153,254 @@ export class Git {
         // if there's an error, assume we're not in a Git working tree
       }
 
-      if (repoInfo && Git.isPathUnderGitWorkingTree(repoInfo)) {
-        Git._gitInfo = repoInfo;
+      if (repoInfo && this.isPathUnderGitWorkingTree(repoInfo)) {
+        this._gitInfo = repoInfo;
       }
-      Git._checkedGitInfo = true;
+      this._checkedGitInfo = true;
     }
-    return Git._gitInfo;
+    return this._gitInfo;
   }
 
-  private static _tryGetGitEmail(): IResultOrError<string> {
-    if (Git._gitEmailResult === undefined) {
-      if (!Git.isGitPresent()) {
-        Git._gitEmailResult = {
-          error: new Error("Git isn't present on the path")
-        };
-      } else {
-        try {
-          Git._gitEmailResult = {
-            result: Utilities.executeCommandAndCaptureOutput('git', ['config', 'user.email'], '.').trim()
-          };
-        } catch (e) {
-          Git._gitEmailResult = {
-            error: e
-          };
+  public getRepositoryRootPath(): string | undefined {
+    const gitPath: string = this.getGitPathOrThrow();
+    const output: child_process.SpawnSyncReturns<string> = Executable.spawnSync(gitPath, [
+      'rev-parse',
+      '--show-toplevel'
+    ]);
+
+    if (output.status !== 0) {
+      return undefined;
+    } else {
+      return output.stdout.trim();
+    }
+  }
+
+  public getChangedFolders(
+    targetBranch: string,
+    skipFetch: boolean = false
+  ): (string | undefined)[] | undefined {
+    if (!skipFetch) {
+      this._fetchRemoteBranch(targetBranch);
+    }
+
+    const gitPath: string = this.getGitPathOrThrow();
+    const output: string = Utilities.executeCommandAndCaptureOutput(
+      gitPath,
+      ['diff', `${targetBranch}...`, '--dirstat=files,0'],
+      this._rushConfiguration.rushJsonFolder
+    );
+    return output.split('\n').map((line) => {
+      if (line) {
+        const delimiterIndex: number = line.indexOf('%');
+        if (delimiterIndex > 0 && delimiterIndex + 1 < line.length) {
+          return line.substring(delimiterIndex + 1).trim();
         }
       }
+
+      return undefined;
+    });
```

Comment:
```suggestion
    }).filter(Boolean);
```
No point leaving `undefined` in the array.